solr/core/src/java/org/apache/solr/core/MMapDirectoryFactory.java

```diff
@@ -61,13 +64,23 @@ public void init(NamedList args) {
   @Override
   protected Directory create(String path, LockFactory lockFactory, DirContext dirContext) throws IOException {
     // we pass NoLockFactory, because the real lock factory is set later by injectLockFactory:
-    MMapDirectory mapDirectory = new MMapDirectory(new File(path).toPath(), lockFactory, maxChunk);
+    final boolean preloadIndexInput = this.preload;
+    MMapDirectory mapDirectory = new MMapDirectory(new File(path).toPath(), lockFactory, maxChunk) {
```

Comment:
I know you don't care about Solr, but this seems to change the system information to show a strange class name instead of the current MMapDirectory output in the system info page of Solr. I have to investigate this, but as said before, I tend to not remove the setPreload() method.